src/d3d8/d3d8_device.cpp

```diff
@@ -496,8 +515,13 @@ namespace dxvk {
           UINT                Height,
           D3DFORMAT           Format,
           IDirect3DSurface8** ppSurface) {
+    // Only CreateImageSurface clears the content of ppSurface
+    // before checking if Format is equal to D3DFMT_UNKNOWN.
     InitReturnPtr(ppSurface);
 
+    if (unlikely(Format == D3DFMT_UNKNOWN))
```

Comment:
What happens on the D3D9 side if the format is UNKNOWN?
Shouldn't we handle this there?

Reply:
D3D9 will always clear the return pointers on all these calls regardless of D3DFMT_UNKNOWN, so it's more of a D3D8 quick.

Reply:
Same thing for all the other instances of this ofc.

Reply:
Oh right. I'd appreciate a bunch of short comments about this.

Reply:
Sure, can do. As you noticed during the previous LockRect/LockBox PRs, D3D8 has an interesting way of doing clears that wasn't preserved in D3D9. Not that I noticed anything caring, but Wine does test for them.